schema/install_schema.go

```diff
@@ -18,7 +18,40 @@ type InstallSchema struct {
 	GrubOptionsSchema   `json:"grub_options,omitempty"`
 	Image               string `json:"image,omitempty" description:"Use a different container image for the installation"`
 	PowerManagement
-	SkipEncryptCopyPlugins bool `json:"skip_copy_kcrypt_plugin,omitempty"`
+	SkipEncryptCopyPlugins bool                `json:"skip_copy_kcrypt_plugin,omitempty"`
+	Partitions             ElementalPartitions `json:"partitions,omitempty" mapstructure:"partitions"`
+	Firmware               string              `json:"firmware,omitempty" mapstructure:"firmware"`
+	PartTable              string              `json:"part-table,omitempty" mapstructure:"part-table"`
+	GrubDefEntry           string              `json:"grub-entry-name,omitempty" mapstructure:"grub-entry-name"`
+	ExtraPartitions        []*Partition        `json:"extra-partitions,omitempty" mapstructure:"extra-partitions"`
+	Force                  bool                `json:"force,omitempty" mapstructure:"force"`
+	ExtraDirsRootfs        []string            `json:"extra-dirs-rootfs,omitempty" mapstructure:"extra-dirs-rootfs"`
+	Active                 Image               `json:"system,omitempty" mapstructure:"system"`
+	Recovery               Image               `json:"recovery-system,omitempty" mapstructure:"recovery-system"`
+	Passive                Image               `json:"passive,omitempty" mapstructure:"recovery-system"`
+}
+
+type Image struct {
+	File       string
+	Label      string `json:"label,omitempty" mapstructure:"label"`
+	Size       uint   `json:"size,omitempty" mapstructure:"size"`
+	FS         string `json:"fs,omitempty" mapstructure:"fs"`
+	Source     string `json:"uri,omitempty" mapstructure:"uri"`
```

Comment:
I have no idea to be honest. The processing of this is done on the spec generation I think so a string should be ok I guess?